### ISSUE TITLE
Test to verify kinetic connection handling and cleanup

### ISSUE THREAD

**Issue by pauldardeau** [swiftstack/kinetic-swift#53]
We need tests to verify that kinetic connection handling and cleanup is performed under various failure scenarios.  We believe that some of the edge cases may need more resiliency.

**Comment by clayg** [comment]
This is a pretty generic issue - it's going to be really hard to close this unless it gets more specific - can we pick a specific daemon or module for *this* issue and open other issues if we find other stuff?

**Comment by pauldardeau** [comment]
Yes, let's focus on replicator only for this issue.  I'll poke around and add comments for potential problem areas.

**Comment by clayg** [comment]
So I've discovered that the drive connection errors only show up when using PUSH mode replication (the kinetic P2P protocol commands).  The issue is not reproducible on the simulator - it only shows up on the drives and may be fixed on newer firmwares.  We'd need to develop some targeted testing against a specific firmware to push the issue forward with Seagate.  It doesn't seem to be related to an issue with kinetic-swift specifically, but may be possible to work around (possibly in conjunction with drive firmware updates) in the python-kinetic bindings.
